plugins/woocommerce/src/Templating/TemplatingEngine.php

```diff
@@ -0,0 +1,807 @@
+				$id
+			);
+
+		return $this->get_rendered_file_core( $sql_query, $include_metadata, $delete_if_expired );
+	}
+
+	/**
+	 * Get information about an existing (as a file and as a database entry) rendered template file.
+	 *
+	 * If no rendered template exists with the specified id, null will be returned.
+	 * Otherwise, an array will be returned with the following keys:
+	 *
+	 * - id (int)
+	 * - file_path (string)
+	 * - date_created_gmt (date, Y-m-d H:m:i)
+	 * - expiration_date_gmt (date, Y-m-d H:m:i)
+	 * - is_public (bool)
+	 * - has_expired (bool)
+	 * - metadata (array), only if $include_metadata is passed as true
+	 *
+	 * The metadata array, if present, will be an associative array of key => value.
+	 *
+	 * IMPORTANT: The physical file pointed by file_path is NOT guaranteed to exist if the rendered template has expired.
+	 * This is due to how the cleanup of expired items works, see delete_expired_rendered_files.
+	 *
+	 * The metadata array, if present, will be an associative array of key => value.
+	 *
+	 * @param string $file_name File name of the rendered template entry.
+	 * @param bool   $include_metadata True to include the rendered template metadata in the result.
+	 * @param bool   $delete_if_expired If true, and the rendered template has expired, it will be deleted (with delete_rendered_file_core) and null will be returned.
+	 * @return array|null Rendered template information array, or null if no entry exists with the supplied id.
+	 * @throws Exception The base directory for rendered files (possibly changed via filter) doesn't exist, or error deleting the file.
+	 */
+	public function get_rendered_file_by_name( string $file_name, bool $include_metadata = false, bool $delete_if_expired = false ): ?array {
+		global $wpdb;
+
+		$sql_query =
+			$wpdb->prepare(
+				"SELECT id, file_name as file_path, date_created_gmt, expiration_date_gmt, is_public FROM {$wpdb->prefix}wc_rendered_templates WHERE file_name=%s",
+				$file_name
+			);
+
+		return $this->get_rendered_file_core( $sql_query, $include_metadata, $delete_if_expired );
+	}
+
+	/**
+	 * Core method to get information about a rendered template file.
+	 *
+	 * @param string $sql_query The SQL query to get the row corresponding to a rendered file id or name.
+	 * @param bool   $include_metadata True to include the rendered template metadata in the result.
+	 * @param bool   $delete_if_expired If true, and the rendered template has expired, it will be deleted (with delete_rendered_file_core) and null will be returned.
+	 * @return array|null Rendered template information array, or null if no entry exists with the supplied id.
+	 * @throws Exception The base directory for rendered files (possibly changed via filter) doesn't exist, or error deleting the file.
+	 */
+	private function get_rendered_file_core( string $sql_query, bool $include_metadata, bool $delete_if_expired ): ?array {
```

Comment:
It doesn't look like this method checks to see if the file actually exists in the filesystem. What happens if there is a database entry for a file, but the file is missing (because of some external file operation or something)?

Reply:
I think the method has a bad name, it should rather be `get_rendered_file_information`: it just returns the (processed) database entry for the file, including the full path where the physical file is supposed to be located. It's the responsibility of the caller to ensure that the file actually exists, which is exactly what the `handle_parse_request` method does when the file is requested via the dedicated REST API endpoint.